core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```diff
@@ -545,3 +511,96 @@ class KRaftMetadataCache(
   }
 }
 
+object KRaftMetadataCache {
```

Comment:
#18632 is trying to remove reference of `KRaftMetadataCache`, so maybe we can move the helpers to `MetadataCache`?